reflex/app.py

```diff
@@ -619,6 +619,24 @@ def _should_compile(self) -> bool:
         return True
 
     def compile(self):
+        """App modules no longer need to call this.
+
+        Reflex framework will perform compile_() as needed.
+
+        Existing call sites will get a deprecation message.
+
+        We will eventually remove app.compile() method entirely.
+        """
+        console.deprecate(
+            feature_name="app.compile()",
+            reason="Explicit calls to app.compile() are not needed."
+            " Method will be removed in 0.4.0",
```

Comment:
This line is not needed, `removal_version` already add this part of the message.